samples/jwt/src/main.c

```diff
@@ -0,0 +1,283 @@
+		snprintf(token_subject_str, APP_JWT_CLAIM_MAX_SIZE, "%s.%s", CONFIG_BOARD,
+			 subject_random_str);
+		token_subject_str[APP_JWT_CLAIM_MAX_SIZE - 1] = '\0';
+	}
+
+	jwt.subject = token_subject_str;
+
+	/* JWT Claims that are not requiered are marked with a `0` */
+	jwt.json_token_id = 0;
+	jwt.audience = 0;
+	jwt.issuer = 0;
+
+	/* Simplified token doesn't need a `iat` claim in its header */
+	jwt.add_timestamp = false,
+
+	/* Call app_jwt API */
+	err = app_jwt_generate(&jwt);
+
+	return err;
+}
+
+static int jwt_generate_full_token(uint32_t exp_time_s, char *out_buffer,
+				   const size_t out_buffer_size)
+{
+	if (!out_buffer || !out_buffer_size) {
+		return -EINVAL;
+	}
+
+	int err = -EINVAL;
+
+	/* Full token used IAK key for signing */
+	struct app_jwt_data jwt = {.sec_tag = 0,
+					.key_type = 0,
+					.alg = JWT_ALG_TYPE_ES256,
+					/* Full token requieres `kid` claim to its header */
+					.add_keyid_to_header = true,
+					/* Full token requieres `iat` claim */
+					.add_timestamp = true,
+					.jwt_buf = out_buffer,
+					.jwt_sz = out_buffer_size};
+
+	/* Full token requieres `exp` claim */
+	if (exp_time_s > APP_JWT_VALID_TIME_S_MAX) {
+		jwt.validity_s = APP_JWT_VALID_TIME_S_MAX;
+	} else if (exp_time_s == 0) {
+		jwt.validity_s = APP_JWT_VALID_TIME_S_DEF;
+	} else {
+		jwt.validity_s = exp_time_s;
+	}
+
+	/* Full token requieres `sub`, `iss`, `jti` and `aud` claims */
+	/* Subject: format: "user_defined_string" , we use uuid as subject */
+	char device_uuid_str[APP_JWT_UUID_V4_STR_LEN] = {0};
+
+	/* Use app_jwt API for UUID */
+	if (0 != app_jwt_get_uuid(device_uuid_str, APP_JWT_UUID_V4_STR_LEN)) {
+		return -ENXIO;
+	}
+
+	jwt.subject = device_uuid_str;
+
+	/* Issuer: format: <hardware_id>.<sub> */
+	char token_issuer_str[APP_JWT_CLAIM_MAX_SIZE] = {0};
+
+	snprintf(token_issuer_str, APP_JWT_CLAIM_MAX_SIZE, "%s.%s", CONFIG_BOARD, jwt.subject);
+	token_issuer_str[APP_JWT_CLAIM_MAX_SIZE - 1] = '\0';
+
+	jwt.issuer = token_issuer_str;
+
+	/* Json Token ID: format: <hardware_id>.<16-random_bytes> */
+	char json_token_id_str[APP_JWT_CLAIM_MAX_SIZE] = {0};
+	char jti_random_str[RANDOM_STR_MAX_SIZE] = {0};
+
+	if (0 == get_random_bytes_str(jti_random_str, RANDOM_STR_MAX_SIZE)) {
+		snprintf(json_token_id_str, APP_JWT_CLAIM_MAX_SIZE, "%s.%s", CONFIG_BOARD,
+			 jti_random_str);
+		json_token_id_str[APP_JWT_CLAIM_MAX_SIZE - 1] = '\0';
+	}
+
+	jwt.json_token_id = json_token_id_str;
+
+	/* Audience: format: "user_defined_string" */
+	char audience_str[APP_JWT_CLAIM_MAX_SIZE] = {0};
+
+	snprintf(audience_str, APP_JWT_CLAIM_MAX_SIZE, "%s", JWT_AUDIENCE_STR);
+	audience_str[APP_JWT_CLAIM_MAX_SIZE - 1] = '\0';
+
+	jwt.audience = audience_str;
+
+	/* Call app_jwt API */
+	err = app_jwt_generate(&jwt);
+
+	return err;
+}
+
+int main(void)
+{
+	LOG_INF("Application JWT sample (%s)", CONFIG_BOARD);
+
+	char jwt_str[APP_JWT_STR_MAX_LEN] = {0};
+
+	LOG_INF("JWT simplified token :");
```

Comment:
A few suggestions to make the sample output look nicer.
```suggestion
	LOG_INF("Generating simplified JWT token");
```